vignettes/tutorial2.Rmd

```diff
@@ -0,0 +1,447 @@
+                                   tvarunconvert_fcn = NULL, 
+                                   pvarunconvert_fcn = exp,
+                                   reconstruction_function = pscl_apply)
+```
+
+Note that the inclusion of the argument `pvarunconvert_fcn = exp` means that the full grids are provided in `(temperature, precipitation)` space. The log(precipitation) transformation has been undone.
+
+
+## The science of each step
+
+### Setup (repeated)
+
+```{r paramsrepeat}
+## parameters for the code below.
+ngen <- 4             # number of fields to generate
+nplt <- ngen          # number of fields to plot
+exfld <- 20           # example field to plot from the time series
+set.seed(8675309)     # Set RNG seed so results will be reproducible
+```
+
+All of the data needed for this tutorial is installed with the package.
+```{r datadirrepeat}
+library('fldgen')
+datadir <- system.file('extdata', package='fldgen')
+```
+
+### Reading the ESM data
+
+The ESM temperature and precipitation fields should be in netCDF filea.  The function that reads it returns a "griddata" structure that contains the data, as well as some information about the grid, such as the latitude, longitude, and time coordinates and the vector of area weights needed to compute grid averages.
+
+```{r readdata}
+filenameT <- file.path(datadir, 'tas_annual_esm_rcp_r2i1p1_2006-2100.nc')
+griddataT <- read.general(filenameT, 
+                          varname='tas', latvar='lat_2', lonvar='lon_2',
+                          timevar='time')
+tgav <- griddataT$vardata %*% griddataT$globalop
+
+
+filenameP <- file.path(datadir, 'pr_annual_esm_rcp_r2i1p1_2006-2100.nc')
+griddataP <- read.general(filenameP, 
+                          varname='pr', latvar='lat', lonvar='lon',
+                          timevar='time')
+```
+Here, we have read in the temperature netCDF data and used the global average operator to compute a time series of global mean temperatures. We have also read in the precipitation netCDF data. These variables will be used as input to the functions that analyze the ESM data and produce the joint temperature and precipitation fields. 
+For the fldgen method to work, each ESM variable must be able to accept residuals of between infinity and -infinity. This is already the case for temperature and no transformation is needed. Because precipitation cannot be less than 0, either the generated residuals have to be constrained to avoid negative precipitation while preserving the ESM spatiotemporal statistics, or the method must operate on a transformation of precipitation that can accept residuals between -infinity and infinity. The latter is more straightforward, and so residuals are generated for log(precipitation) rather than precipitation. Any function that is continuous, invertible, and monotonic and results in a transformed variable supported on (-infinity, infinity) will preserve the ESM spatiotemporal statistical properties as desired. 
```

Comment:
This is all explained in the one-step section.  No need to repeat it here, I think.
```suggestion
Next we must define the transformation functions, as described above.  Temperature doesn't need one,
but for precipitation we will use a log transformation so that we can ensure that output precipitation value
are >= 0.
```